crates/sncast/src/main.rs

```diff
@@ -331,9 +353,18 @@ async fn run_async_command(
             let selector = get_selector_from_name(&function)
                 .context("Failed to convert entry point selector to FieldElement")?;
 
+            let class_hash = get_class_hash_by_address(&provider, contract_address)
+                .await
+                .with_context(|| format!("Couldn't retrieve class hash of a contract with address {contract_address:#x}"))?
+                .with_context(|| format!("Couldn't retrieve class hash of a contract with address {contract_address:#x}"))?;
```

Comment:
This is duplicated

Reply:
Indeed. `get_class_hash_by_address` returns `Result<Option<Felt>>`. We need to handle both `Err` and `None`